### PR TITLE
Added UI for Update Info page

### LINKED ISSUE #114: [FEATURE] Add UI for Update Info page
### Description

Currently the Update Info page has no UI and functionality. Design to be discussed with team and implemented.
https://github.com/warmachine028/uemcrp/issues/105#issuecomment-1296215863

### Screenshots

image

image

### Additional information

- Always use styled componets instead of plain CSS and MUI components.
## Happy contributing. 💐
Star my other Repositories [here](https://github.com/warmachine028)

### PR DESCRIPTION
**Pull request by anuj98**
# Issue  📐

- **I anuj98 have worked for #114**

[put x to check the boxes]: <> (This is a comment, it will not be included)

## Guidelines 🔐

**I accept the fact that i have followed the guidelines and have not copied the codes from around the internet**

- [x] **Contribution Guidelines**
- [x] **Code of Conduct**

## Issue to be closed 🛅

- **My pull request closes #114**

## Screenshots 📷

**Here are the pictures of changes that i have made 🔽**

Laptop:
image

Mobile:
image

  

---

### CODE REVIEW

File: client/src/components/studentModules/UpdateInfo/index.jsx [hunk 1]
```diff
@@ -1,6 +1,93 @@
...
+const UpdateInfo = ({ user, updateUserInfo }) => {
+	const [email, setEmail] = useState(user.email);
+	const [phoneNumber, setPhoneNumber] = useState(user.mobile);
+	const [address, setAddress] = useState(user.permanentAddress);
+
+	const handleChange = (event, type) => {
+		switch (type) {
+			case InputType.Email: setEmail(event.target.value); break;
+			case InputType.Phone: setPhoneNumber(event.value); break;
+			case InputType.HouseFlat: setAddress({ ...address, houseFlat: event.target.value }); break;
```
Comment (anuj98) [inline client/src/components/studentModules/UpdateInfo/index.jsx:27]: event.target.value

File: client/src/components/studentModules/UpdateInfo/index.jsx [hunk 2]
```diff
@@ -1,6 +1,93 @@
...
+
+const UpdateInfo = ({ user, updateUserInfo }) => {
+	const [email, setEmail] = useState(user.email);
+	const [phoneNumber, setPhoneNumber] = useState(user.mobile);
+	const [address, setAddress] = useState(user.permanentAddress);
+
+	const handleChange = (event, type) => {
+		switch (type) {
+			case InputType.Email: setEmail(event.target.value); break;
+			case InputType.Phone: setPhoneNumber(event.value); break;
```
Comment (anuj98) [inline client/src/components/studentModules/UpdateInfo/index.jsx:26]: event.target.value

File: client/src/App/index.jsx [hunk 3]
```diff
@@ -21,6 +21,10 @@ const App = () => {
 	const test = null || students[0]
 	const [user, setUser] = useState(test)
 
+	const updateUserInfo = user => {
```
Comment (anuj98) [inline client/src/App/index.jsx:24]: No need of this function. Pass setUser directly to props

**Comment by warmachine028** [comment]
The Email Address input field is missing, I guess, could you have a look?

**Pull request merged**